### PR TITLE
Add tutorial for static resources example project

### LINKED ISSUE #106: Write up tutorial for static resources example
Since #105 was merged, we need to write up a proper tutorial in a `README.md` which builds upon concepts introduced from the previous examples and also change the `2-bigger-project` tutorial text to reference this new example project.

### PR DESCRIPTION
**Pull request by ebkalderon**
### Added

* Add tutorial for static resources example project.

### Changed

* Update end of `README.md` for `2-bigger-project` for proper continuity.
* Rename `3-project-with-resources` to `3-static-resources`.
* Fix indentation and simplify `localPatterns` expression in `3-static-resources`.
* Remove auto-generated comments from `Cargo.toml` files in examples.
* Preemptively reference `v0.9.0` tag in tutorials.
* Update example projects' `cargo2nix` pins to a newer version compatible with #119.

Closes #106.

### CODE REVIEW

File: examples/1-hello-world/README.md [hunk 1]
```diff
@@ -127,7 +127,7 @@ with the following arguments:
   },
   cargo2nix ? builtins.fetchGit {
     url = https://github.com/tenx-tech/cargo2nix;
-    ref = "v0.8.3";
+    ref = "v0.9.0";
```
Comment (antigravityla) [inline examples/1-hello-world/README.md:130]: https://github.com/tenx-tech/cargo2nix/blob/master/Cargo.toml#L3

Reminder, need version bump to show up

File: examples/3-project-with-resources/README.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    Ructe::from_env()
+        .expect("ructe")
+        .compile_templates("templates")
+        .unwrap();
+}
+```
+
+This build script will generate a Rust source file named `$OUT_DIR/templates.rs`
+which we can statically embed into our program with `include_str!()`. Let's open
+the `src/main.rs` file eand write some code that will do just that:
```
Comment (antigravityla) [inline examples/3-project-with-resources/README.md]: nit `eand`

File: examples/3-static-resources/README.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+│   └── main.rs
+├── tests
+│   └── my_tests.rs
+├── build.rs
+├── default.nix
+├── Cargo.lock
+├── Cargo.nix
+├── Cargo.toml
+├── README.md
+└── result -> /nix/store/qjz1m7pmm23r7fhgbigb2x5pnwqp925j-crate-foo-project-0.1.0
```
[the remark this answers is not in the document]
Comment (ebkalderon) [reply examples/3-static-resources/README.md:29]: Yeah, you're probably right. I was on the fence about whether to keep this or not, but I wanted to make sure it was 100% understandable to the reader. If you, as a reader think it's clear enough without it, I'll gladly trim this part out. 👍

[the remark this answers is not in the document]
Comment (ebkalderon) [reply examples/3-static-resources/README.md:29]: Yeah, you're probably right. I was on the fence about whether to keep this or not, but I wanted to make sure it was 100% understandable to the reader. If you, as a reader think it's clear enough without it, I'll gladly trim this part out.

File: examples/1-hello-world/README.md (hunk 1 above)
Comment (ebkalderon) [reply examples/1-hello-world/README.md:130]: Yep, thanks for the reminder! I decided to preemptively reference this tag in the tutorials in preparation for the v0.9.0 release (which should hopefully be soon, currently blocked on #129 being resolved). The idea is to have the tutorials ready in `master` for the new release before we apply the `v0.9.0` tag and publish it on GitHub.

File: examples/3-project-with-resources/README.md (hunk 2 above)
Comment (ebkalderon) [reply examples/3-project-with-resources/README.md]: Whoops! Good eye there, @antigravityla. Will fix. ❤️

**Comment by ebkalderon** [comment]
Thanks for the reviews, everyone! Given the number of approvals already and that GitHub will invalidate them as soon as I push another commit, I'll merge this PR first for convenience and then open a follow-up PR to address the suggestions @antigravityla provided.

**Pull request merged**